core/dbt/cli/context.py

```diff
@@ -0,0 +1,39 @@
+from click import Context, Group, Command
+from click.exceptions import NoSuchOption, UsageError
+from dbt.config.runtime import load_project, load_profile
+from dbt.cli.flags import Flags
+import sys
+
+
+class DBTUsageException(Exception):
+    pass
+
+
+class DBTContext(Context):
+    def __init__(self, command: Command, **kwargs) -> None:
+        invocation_args = kwargs.pop("args", sys.argv[1:])
+        super().__init__(command, **kwargs)
+
+        # Bubble up validation errors for top-level commands
+        if not self.parent:
+            self._validate_args(command, invocation_args)
+
+        if not self.obj:
+            flags = Flags(self, args=invocation_args)
```

Comment:
How do you imagine this work with the current setup stuff we do in `cli` group?

Reply:
Removed it from the `cli` group method, instead the runner sets `cli.context_class = DBTContext` here: https://github.com/dbt-labs/dbt-core/pull/6387/files#diff-43832707bfec8cdaa49c13f004d9b3296585827f7e68ef07fba7ad1d5e6eb7c6R24